### PR TITLE
Fix Session Timeout Message Duplication

### LINKED ISSUE #789: [Timed out session] Two different warning messages appear
**Environment**: Windows 10 Pro 22H2 Google Chrome Version 120.0.6099.225 (Official Build) (64-bit)
**Reproducible**: when session time ends

**Steps to reproduce**

1. Log in as Admin
2. After non-working time check if warning message appears

**Actual result**
Two different warning messages 'true' and 'Час вашої сесії вичерпано. Будь ласка, увійдіть до системи знову' appear

**Postconditions**
One warning message 'Час вашої сесії вичерпано. Будь ласка, увійдіть до системи знову' should appear

User story and test case links
E.g.: "User story 

image

### PR DESCRIPTION
**Pull request by Natali-Kotelnitska**
dev

* resolves #789 

## Code reviewers

- [ ] @loqimean 
- [x] @obniavko 

## Summary of issue

There is currently an issue where two different warning messages, 'true' and "Your session expired. Please sign in again to continue", are appearing after non-working time when logged in. This inconsistency in warning messages needs to be resolved.

#### flash messages (before)
image

## Summary of change

The issue was fixed by modifying the flash message iteration in the view. The 'timedout' message type is now skipped, preventing the 'true' message from being displayed.

#### flash message (after changes)
Знімок екрана (608)

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by obniavko — APPROVED** [review]

File: app/views/layouts/_flash_messages.html.erb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 <% flash.each do |message_type, message| %>
+  <% next if message_type == 'timedout' %>
```
Comment (loqimean) [inline app/views/layouts/_flash_messages.html.erb:2]: а якщо будуть інші дублікати? не окей, бо тоді ти будеш додавати так до безкінечності, ти маєш задати універсальне рішення

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/views/layouts/_flash_messages.html.erb (hunk 1 above)
Comment (Natali-Kotelnitska) [reply app/views/layouts/_flash_messages.html.erb]: Переробила.

File: app/views/layouts/_flash_messages.html.erb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 <% flash.each do |message_type, message| %>
+  <% next unless message.is_a? String %>
```
Comment (loqimean) [inline app/views/layouts/_flash_messages.html.erb:2]: а не простіше не ітеруватися по кейсам, коли нема месиджів?)

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/views/layouts/_flash_messages.html.erb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <% flash.each do |message_type, message| %>
-  <%= turbo_stream.toast(message, background: message_type) %>
+  <%= turbo_stream.toast(message, background: message_type) if message.is_a? String %>
```
Comment (loqimean) [inline app/views/layouts/_flash_messages.html.erb:2]: так а чому так? які ще кейси бувають? масив?)

File: app/views/layouts/_flash_messages.html.erb [hunk 4]
```diff
@@ -1,3 +1,3 @@
-<% flash.each do |message_type, message| %>
-  <%= turbo_stream.toast(message, background: message_type) %>
+<% flash.to_h.excluding("timedout").each do |message_type, message| %>
```
Comment (loqimean) [inline app/views/layouts/_flash_messages.html.erb:1]:
```suggestion
<% flash.to_h.except("timedout").each do |message_type, message| %>
```

Comment (loqimean) [reply app/views/layouts/_flash_messages.html.erb:1]: краще це заюзай і можеш мерджати

**Review by loqimean — APPROVED** [review]

**Pull request merged**